src/wam2layers/examples.py

```diff
@@ -0,0 +1,58 @@
+"""Functions for downloading example cases."""
+
+import io
+import zipfile
+from pathlib import Path
+from typing import Callable
+
+import requests
+
+
+def extract_from_4TU_archive(download_link, target_dir):
+    # See https://stackoverflow.com/a/14260592
+    response = requests.get(download_link, stream=True)  # TODO stream necessary or not?
```

Comment:
Reading response.content creates a bytes var of whole zipfile in memory.
To reduce memory footprint you could be use something like
```python
import requests
import zipfile
import tempfile
import shutil

url = 'http://localhost:8000/run1-full.zip'
# url = 'http://localhost:8000/docking-protein-protein-full-local.zip'
target_dir = 'bla'
with requests.get(url, stream=True) as response:
    with tempfile.TemporaryFile() as f:
        shutil.copyfileobj(response.raw, f)
        f.seek(0)
        with zipfile.ZipFile(f) as z:
            z.extractall(target_dir)
```

context on get() is needed to close connection
copyfileobj() is needed because response.raw is a file like object but cant seek, which is needed by ziplfile.

Reply:
Instead of TemporaryFile you could use BytesIO if file is always small or SpooledTemporaryFile if you want small file to  not be written to disk and big file should.

Reply:
Thanks! I think I get most of it now :-)
In the end I did a file-by-file download, but this will surely be useful in the future